android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt

```diff
@@ -0,0 +1,111 @@
+     * iOS only, unused on Android.
+     * @param promise to resolve and return the results.
+     */
+    @ReactMethod
+    fun showPromptForPushNotifications(pushConfigurationOptions: ReadableMap?, promise: Promise) {
+        // Skip requesting permissions when already granted
+        if (checkPushPermissionStatus() == PermissionStatus.Granted) {
+            promise.resolve(PermissionStatus.Granted.toReactNativeResult)
+            return
+        }
+
+        try {
+            notificationRequestPromise = promise
+            (currentActivity as PermissionAwareActivity).requestPermissions(
+                arrayOf(POST_NOTIFICATIONS_PERMISSION_NAME),
+                POST_NOTIFICATIONS_PERMISSION_REQUEST,
+                this,
+            )
+        } catch (ex: Throwable) {
+            promise.reject(ex)
+            notificationRequestPromise = null
+        }
+    }
+
+    /**
+     * Checks current permission of push notification permission
+     */
+    private fun checkPushPermissionStatus(): PermissionStatus =
+        // Skip requesting permissions for older versions where not required
+        if (Build.VERSION.SDK_INT < BUILD_VERSION_CODE_TIRAMISU || ContextCompat.checkSelfPermission(
+                reactContext, POST_NOTIFICATIONS_PERMISSION_NAME,
+            ) == PackageManager.PERMISSION_GRANTED
+        ) PermissionStatus.Granted
+        else PermissionStatus.Denied
+
+    /**
+     * Resolves and clears promise with the provided permission status
+     */
+    private fun resolvePermissionPromise(status: PermissionStatus) {
+        notificationRequestPromise?.resolve(status.toReactNativeResult)
+        notificationRequestPromise = null
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int, permissions: Array<String>, grantResults: IntArray,
+    ): Boolean = when (requestCode) {
+        POST_NOTIFICATIONS_PERMISSION_REQUEST -> {
+            // If request is cancelled, the result arrays are empty.
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                resolvePermissionPromise(PermissionStatus.Granted)
+            } else {
+                resolvePermissionPromise(PermissionStatus.Denied)
+            }
+            true // as this permission listener can be removed now
+        }
+        else -> false // desired permission not yet granted, so we will keep the listener
+    }
+
+    override fun getName(): String = "CustomerioPushMessaging"
+
+    /**
+     * Maps native class to react native supported type so the result can be passed on to JS/TS classes.
+     */
+    private val PermissionStatus.toReactNativeResult: Any
+        get() = this.name
+
+    companion object {
+        /**
+         * Copying value os [Manifest.permission.POST_NOTIFICATIONS] as constant so we don't have to
+         * force newer compile sdk versions
+         */
+        private const val POST_NOTIFICATIONS_PERMISSION_NAME =
```

Comment:
Pretty sure we can replace this with a [reference to the permission string in the Android SDK](https://developer.android.com/reference/android/Manifest.permission#POST_NOTIFICATIONS)? 

I understand that the string was added in API 33. But I think that we should be checking for the Android SDK version in any function that uses this constant anyway. We currently check the SDK version in `checkPushPermissionStatus` but we don't in `showPromptForPushNotifications`. 

What if we were to refactor `showPromptForPushNotifications` to first check for Android SDK version, return early and ignore the request if the Android SDK is too low. Then, we can remove this constant and make a reference `Manifest.permission.POST_NOTIFICATIONS`?

Reply:
No. We cannot replace it with permission string in the Android SDK for the reason explained in comment above it. Anything that was added in API Level 33 is unknown to earlier API Levels. So if any app is using `compileSdkVersion` less than `33`, they will fail on compiling with following errors:

> Unresolved reference: POST_NOTIFICATIONS
> Unresolved reference: TIRAMISU

For the check of API 33, we [already do it in `showPromptForPushNotifications` by calling `checkPushPermissionStatus` with early return](104/files#diff-74884e5e24c17161d56be01a332a64aeabd1f5bbafefab2bbd409befbb72fdd6R40-R44).